app/models/semester.rb

```diff
@@ -23,11 +22,7 @@ def can_be_destroyed?
     ! (student_applications.exists? || projects.exists?)
   end
 
-  def self.current_semester
-    where(is_current_semester: true).first
-  end
-
-  def self.clear_current_semester
-    update_all(is_current_semester: false)
+  def season_with_year
+    "#{season} #{year}"
   end
```

Comment:
Is there a reason not to override the `to_s` method instead of making a new one?